lib/util/Components.js

```diff
@@ -255,9 +255,24 @@ function componentRule(rule, context) {
         node[property] &&
         node[property].type === 'JSXElement'
       ;
+      var destructuredReactCreateElement = function () {
+        var variables = variableUtil.variablesInScope(context);
+        var variable = variableUtil.getVariable(variables, 'createElement');
+        if (variable !== null) {
+          var map = variable.scope.set;
+          // eslint-disable-next-line no-undef
+          if (map instanceof Map && map.has('React')) {
```

Comment:
There's no need to use Map if the keys are strings; you can just use a normal object.